src/main/java/spring/intro/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package spring.intro.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import spring.intro.dao.UserDao;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    private UserDao userDao;
```

Comment:
What is better: setter, constructor, or field injection?

Reply:
constructor. changed.